runtime-modules/forum/src/lib.rs

```diff
@@ -503,13 +503,19 @@ decl_storage! {
         pub NextCategoryId get(next_category_id) config(): T::CategoryId;
 
         /// Map thread identifier to corresponding thread.
-        pub ThreadById get(thread_by_id) config(): map T::ThreadId => Thread<T::ForumUserId, T::CategoryId, T::Moment, T::Hash>;
+        pub ThreadById get(thread_by_id) config(): double_map T::CategoryId, blake2_256(T::ThreadId) => Thread<T::ForumUserId, T::CategoryId, T::Moment, T::Hash>;
+
+        // Map category to thread. Enables searching for threads in ThreadById without knowing to which category thread belongs to.
+        pub CategoryByThread get(category_by_thread) config(): map T::ThreadId => T::CategoryId;
 
         /// Thread identifier value to be used for next Thread in threadById.
         pub NextThreadId get(next_thread_id) config(): T::ThreadId;
 
         /// Map post identifier to corresponding post.
-        pub PostById get(post_by_id) config(): map T::PostId => Post<T::ForumUserId, T::ThreadId, T::Hash>;
+        pub PostById get(post_by_id) config(): double_map T::ThreadId, blake2_256(T::PostId) => Post<T::ForumUserId, T::ThreadId, T::Hash>;
+
+        // Map thread to post. Enables searching for posts in PostById without knowing to which thread post belongs to.
+        pub ThreadByPost get(thread_by_post) config(): map T::PostId => T::ThreadId;
```

Comment:
Why was `ThreadByPost get(thread_by_post) config(): map T::PostId => T::ThreadId` introduced? a post already explicitly says what thread it belongs to by `Post::thread_id`.

Reply:
If the problem is that you are unable to lookup a post by only its ID after the introduction of the double map, then the proper solution is to require the user to provide a combination of the post id and thread id when trying to fully identify a thread.